client/starwhale/utils/fs.py

```diff
@@ -42,8 +43,10 @@ def ensure_file(
             raise
 
     if _saved != content or not p.exists():
-        # TODO: add timestamp for tmp file
-        _tmp_f = p.parent / f".{p.name}.tmp"
+        # add random num to tmp file name to avoid conflict
+        _tmp_f = p.parent / f".{p.name}.tmp.{random.randint(0, 100000)}"
```

Comment:
could we use `tempfile` to create a named random temp file in the p.parent dir?

Reply:
Done